parmed/openmm/parameters.py

```diff
@@ -122,18 +141,180 @@ def write(self, dest, provenance=None):
             own_handle = False
 
         try:
-            if provenance is not None:
-                self._write_omm_provenance(provenance)
+            dest.write('<ForceField>\n')
+            self._write_omm_provenance(dest, original_file, reference)
+            self._write_omm_atom_types(dest)
             self._write_omm_residues(dest)
             self._write_omm_bonds(dest)
             self._write_omm_angles(dest)
-            self._write_omm_ubs(dest)
             self._write_omm_dihedrals(dest)
-            self._write_omm_periodic_impropers(dest)
             self._write_omm_impropers(dest)
-            self._write_omm_rb_torsions(dest)
+#           self._write_omm_rb_torsions(dest)
             self._write_omm_cmaps(dest)
-            self._write_omm_nonbonded(dest)
+#           self._write_omm_scripts(dest)
+#           self._write_omm_nonbonded(dest)
         finally:
+            dest.write('</ForceField>\n')
             if own_handle:
                 dest.close()
+
+    def _write_omm_provenance(self, dest, original_file, reference):
+        dest.write(' <Info>\n')
+        dest.write('  <Source>%s</Source>\n' % original_file)
+        dest.write('  <DateGenerated>%02d-%02d-%02d</DateGenerated>\n' %
+                   datetime.datetime.now().timetuple()[:3])
+        dest.write('  <Reference>%s</Reference>\n' % reference)
+        dest.write(' </Info>\n')
+
+    def _write_omm_atom_types(self, dest):
+        dest.write(' <AtomTypes>\n')
+        for name, atom_type in iteritems(self.atom_types):
+            assert atom_type.atomic_number >= 0, 'Atomic number not set!'
+            element = Element[atom_type.atomic_number]
+            dest.write('  <Type name="%s" class="%d" element="%s" mass="%f"/>\n'
+                % (name, atom_type.number, element,
+                   atom_type.mass or Mass[element])
+            )
+        dest.write(' </AtomTypes>\n')
+
+    def _write_omm_residues(self, dest):
+        self.typeify_templates()
+        dest.write(' <Residues>\n')
+        for name, residue in iteritems(self.residues):
+            if not isinstance(residue, ResidueTemplate):
+                continue
+            dest.write('  <Residue name="%s">\n' % residue.name)
+            for atom in residue.atoms:
+                dest.write('   <Atom name="%s" type="%s" charge="%f"/>\n' %
```

Comment:
What is your proposed change?  I picked this because I didn't think it mattered how much precision I printed stuff to in the XML file...